control/grpc.py

```diff
@@ -236,41 +266,173 @@ def __init__(self, config: GatewayConfig, gateway_state: GatewayStateHandler, rp
         self.subsys_max_ns = {}
         self.host_info = SubsystemHostAuth()
 
-    def create_host_psk_file(self, subsysnqn : str, hostnqn : str, psk_value : str) -> str:
+    def get_directories_for_key_file(self, key_type : str, subsysnqn : str, create_dir : bool = False) -> []:
+        tmp_dirs = []
+        dir_prefix = f"{key_type}_{subsysnqn}_"
+
+        try:
+            for f in pathlib.Path("/tmp").iterdir():
+                if not os.path.isdir(f):
+                    continue
+                if not f.parts[2].startswith(dir_prefix):
+                    continue
+                tmp_dirs.insert(0, str(f))
```

Comment:
Is it not easier (to read) if you express this as a positive condition (and stick to `pathlib`, instead of using `os.path` as well?):
```suggestion
            for f in pathlib.Path("/tmp").iterdir():
                if f.is_dir() and f.name.startswith(dir_prefix):
                    tmp_dirs.insert(0, str(f))
```
Or even better, with a list comprehension:
```suggestion
            return [str(f) for f in pathlib.Path("/tmp").iterdir()
                    if f.is_dir() and f.name.startswith(dir_prefix)]
```

Reply:
I use the negative conditions with continue when there a lot of them. I guess here I don't need it as there are only two conditions. I still prefer to use the insert. It's easier for me to follow the logic that way.